### PR TITLE
Improve test suite

### PR DESCRIPTION
**Pull request by edoardocavazza**
**Wiki**: https://github.com/Chialab/rna-cli/wiki/Testing-with-SauceLabs

**Features**
* Introduce `--targets` which accepts a `browserlist` query.
* Convert `browserslist`'s queries to SauceLabs VMs.

**Refactor**
* Use `Entry` module to detect packages.
* Use `nightwatch` local bin instead of library.
* Improve SauceLabs compatibility.
* `unit --server` does not automatically launch browsers.
* `unit --browser` launches all locally installed browsers.

**Fixes**
* Align RealFaviconGenerator apis (outsider).
* Generic issues.

### CODE REVIEW

File: commands/build/action.js [hunk 1]
```diff
@@ -46,119 +47,121 @@ module.exports = (app, options = {}, profiler) => {
...
-                } else if (styleOptions.output && ext.isStyleFile(json.main)) {
-                    // if output option is different from the main field
-                    // we can use the main file as source if it is a style.
-                    styleOptions.input = path.join(pkg.path, json.main);
-                }
-                if (styleOptions.input) {
-                    // a style source has been detected.
-                    packageBundlePromise = packageBundlePromise.then(() =>
-                        sass(app, styleOptions, profiler)
+                    opts.targets = opts.targets ? browserslist.elaborate(opts.targets) : browserslist.load(opts.input);
```
[the remark this answers is not in the document]
Comment (edoardocavazza) [reply commands/build/action.js]: `opts.input` has been set above, at line ~60

File: commands/documentation/action.js [hunk 2]
```diff
@@ -67,33 +66,36 @@ module.exports = (app, options) => {
...
+            let output = path.resolve(entry.package.path, options.output);
+            if (!path.extname(output)) {
+                // `output` options is a folder
+                let shortName = entry.package.name.split('/')[1];
+                // generate a file with the name of the package.
+                output = path.join(output, `${shortName.toLowerCase()}.md`);
+            }
+            promise = promise.then(() => generate(
+                app,
+                [path.join(entry.package.path, entry.package.json.module || entry.package.json.main)],
```
[the remark this answers is not in the document]
Comment (edoardocavazza) [reply commands/documentation/action.js]: `package.path` and `package.json` are mandatory properties of the `PackageEntry`, we are confident that those properties exist.

**Comment by nicolocarpignoli** [comment]
@edoardocavazza please add a check in `e2e` command that log an error or similar if are missing USERNAME or ACCESS KEY variables

File: commands/build/action.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (nicolocarpignoli) [reply commands/build/action.js]: ok thanks

**Comment by edoardocavazza** [comment]
@nicolocarpignoli done!

**Comment by nicolocarpignoli** [comment]
@edoardocavazza It would be also nice to list valid entries for query in `--targets <query>` option for e2e command, in this wiki https://github.com/Chialab/rna-cli/wiki/Testing-with-SauceLabs

When you have done this please tell me and I will merge :)

**Pull request merged**